hydra-web/src/js/app.js

```diff
@@ -119,17 +129,17 @@ function(
             var urlPath;
             if (tokenName == "token") {
                 urlPath = "/authentication/login";
-            } else if (tokenName == "sudo") {
-              urlPath = "/authentication/sudo";
             } else {
                 alertify.error("Unknown token name " + tokenName);
                 return;
             }
             var username = $.trim(usernameInput.value);
             var password = passwordInput.value;
+            passwordCache = passwordInput.value;
```

Comment:
You should be using a secure https only cookie that cannot be accessed by js and that gets refreshed on every request so that it doesn't expire.